zeroex/orderwatch/order_watcher.go

```diff
@@ -1165,32 +1229,37 @@ func (w *Watcher) findOrder(orderHash common.Hash) *types.OrderWithMetadata {
 // findOrdersByTokenAddressAndTokenID finds and returns all orders that have
 // either a makerAsset or a makerFeeAsset matching the given tokenAddress and
 // tokenID.
-func (w *Watcher) findOrdersByTokenAddressAndTokenID(makerAddress, tokenAddress common.Address, tokenID *big.Int) ([]*types.OrderWithMetadata, error) {
-	ordersWithAffectedMakerAsset, err := w.db.FindOrders(&db.OrderQuery{
-		Filters: []db.OrderFilter{
-			{
-				Field: db.OFMakerAddress,
-				Kind:  db.Equal,
-				Value: makerAddress,
-			},
-			db.MakerAssetIncludesTokenAddressAndTokenID(tokenAddress, tokenID),
+func (w *Watcher) findOrdersByTokenAddressAndTokenID(makerAddress, tokenAddress common.Address, tokenID *big.Int, filter db.OrderFilter) ([]*types.OrderWithMetadata, error) {
+	filters := []db.OrderFilter{
+		{
+			Field: db.OFMakerAddress,
+			Kind:  db.Equal,
+			Value: makerAddress,
 		},
-	})
+		db.MakerAssetIncludesTokenAddressAndTokenID(tokenAddress, tokenID),
+	}
+	if filter.Kind != "" {
```

Comment:
`filter.Kind` should never be `""` for any valid filter. If your intention is to express optionality for the `filter` argument, maybe we should just make it a pointer (which can be nil) or a slice (which can be empty)?